clients/unshaded/src/main/java/tachyon/hadoop/AbstractTFS.java

```diff
@@ -423,10 +422,12 @@ public boolean mkdirs(Path cPath, FsPermission permission) throws IOException {
     TachyonURI path = new TachyonURI(Utils.getPathWithoutScheme(cPath));
     MkdirOptions options = new MkdirOptions.Builder(mTachyonConf).setRecursive(true).build();
     try {
+      TachyonFile fileId = mTFS.openIfExists(path);
+      if (fileId != null && mTFS.getInfo(fileId).isIsFolder()) {
+        // The directory already exists, nothing to do here
+        return true;
+      }
```

Comment:
We could change our API back to being the same as Hadoop's, and report success on creating an already-existing directory. I prefer this though since mkdirs` performance is unlikely to be important.